Gems/Atom/Feature/Common/Code/Source/Mesh/ModelReloader.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <RenderCommon.h>
+#include <AzCore/Asset/AssetCommon.h>
+#include <AzCore/RTTI/TypeInfo.h>
+#include <AzCore/std/containers/unordered_set.h>
+#include <Atom/Feature/Mesh/ModelReloaderSystemInterface.h>
+#include <Source/Mesh/ModelReloaderSystem.h>
+
+namespace AZ
+{
+    namespace RPI
+    {
+        class ModelAsset;
+    }
+
+    namespace Render
+    {
+        //! ModelReloader takes care of reloading Buffer, ModelLod, and Model assets in the correct order
+        //! The ModelReloaderSystem should be used to reload a model, rather than using a ModelReloader directly
+        class ModelReloader
+            : private Data::AssetBus::MultiHandler
+        {
+            using DependencyList = AZStd::vector<Data::Asset<Data::AssetData>>;
```

Comment:
Would this be more efficient as an unordered set?  You might need to pass in a hash function as a template parameter that uses the asset ID.

Reply:
More efficient when adding the assets, but less efficient when looking up the index within the set to determine which bit in the bitmask needs to be marked as reloaded.